realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.kt

```diff
@@ -0,0 +1,784 @@
+
+    val backlinkFields: Set<Backlink>
+        get() = backlinks.toSet()
+
+    val primaryKeyGetter: String
+        get() = getInternalGetter(primaryKey!!.simpleName.toString())
+
+    /**
+     * Returns `true if the class is considered to be a valid RealmObject class.
+     * RealmObject and Proxy classes also have the @RealmClass annotation but are not considered valid
+     * RealmObject classes.
+     */
+    val isModelClass: Boolean
+        get() {
+            val type = classType.toString()
+            return type != "io.realm.DynamicRealmObject" && !type.endsWith(".RealmObject") && !type.endsWith("RealmProxy")
+        }
+
+    val classElement: Element
+        get() = classType
+
+    init {
+        typeUtils = env.typeUtils
+        elements = env.elementUtils
+
+
+        validPrimaryKeyTypes = Arrays.asList(
+                typeMirrors.STRING_MIRROR,
+                typeMirrors.PRIMITIVE_LONG_MIRROR,
+                typeMirrors.PRIMITIVE_INT_MIRROR,
+                typeMirrors.PRIMITIVE_SHORT_MIRROR,
+                typeMirrors.PRIMITIVE_BYTE_MIRROR
+        )
+
+        validListValueTypes = Arrays.asList(
+                typeMirrors.STRING_MIRROR,
+                typeMirrors.BINARY_MIRROR,
+                typeMirrors.BOOLEAN_MIRROR,
+                typeMirrors.LONG_MIRROR,
+                typeMirrors.INTEGER_MIRROR,
+                typeMirrors.SHORT_MIRROR,
+                typeMirrors.BYTE_MIRROR,
+                typeMirrors.DOUBLE_MIRROR,
+                typeMirrors.FLOAT_MIRROR,
+                typeMirrors.DATE_MIRROR
+        )
+
+        for (element in classType.enclosedElements) {
+            if (element is ExecutableElement) {
+                val name = element.getSimpleName()
+                when {
+                    name.contentEquals("toString") -> this.containsToString = true
+                    name.contentEquals("equals") -> this.containsEquals = true
+                    name.contentEquals("hashCode") -> this.containsHashCode = true
+                }
+            }
+        }
+
+        ignoreKotlinNullability = java.lang.Boolean.valueOf(
+                (env.options as MutableMap<String, String>).getOrDefault(OPTION_IGNORE_KOTLIN_NULLABILITY, "false"))
+    }
+
+    override fun toString(): String {
+        return "class $qualifiedClassName"
+    }
+
+    /**
+     * Returns the internal field name that matches the one in the Java model class.
+     */
+    fun getInternalFieldName(javaFieldName: String): String {
+        for (field in fields) {
+            if (field.javaName == javaFieldName) {
+                return field.internalFieldName
+            }
+        }
+        throw IllegalArgumentException("Could not find fieldname: $javaFieldName")
+    }
+
+    /**
+     * Returns all persistable fields that reference other Realm objects.
+     */
+    fun getObjectReferenceFields(): List<RealmFieldElement> {
+        return Collections.unmodifiableList(objectReferenceFields)
```

Comment:
I think just do `objectReferenceFields.toList()` will be more kotlin here.
Or even this getter can be removed, use backing field like:
```
    val foo : List<Int>
        get() = _foo.toList()

    private val _foo = mutableListOf<Int>()
```

Reply:
Chen! You're back!!!!!!